cmd/livepeer/starter/starter.go

```diff
@@ -1494,6 +1524,73 @@ func parseEthKeystorePath(ethKeystorePath string) (keystorePath, error) {
 	return keystore, nil
 }
 
+func parsePricePerUnit(pricePerUnitStr string) (*big.Rat, string, error) {
+	match := pricePerUnitRex.FindStringSubmatch(pricePerUnitStr)
+	if match == nil {
+		return nil, "", fmt.Errorf("price must be in the format of <price><currency>, provided %v", pricePerUnitStr)
+	}
+	price, currency := match[1], match[3]
+
+	pricePerUnit, ok := new(big.Rat).SetString(price)
+	if !ok {
+		return nil, "", fmt.Errorf("price must be a valid number, provided %v", match[1])
+	} else if pricePerUnit.Sign() < 0 {
+		return nil, "", fmt.Errorf("price must be >= 0, provided %v", pricePerUnit)
+	}
+	if currency == "" {
+		currency = "wei"
+	}
+
+	return pricePerUnit, currency, nil
+}
+
+func startPriceUpdateLoop(ctx context.Context, rpcURL, priceFeedAddr string, pricePerUnit *big.Rat, currency string, updatePrice func(*big.Rat)) error {
+	if strings.ToLower(currency) == "wei" {
+		updatePrice(pricePerUnit)
+		return nil
+	} else if strings.ToUpper(currency) == "ETH" {
+		weiPrice := new(big.Rat).Mul(pricePerUnit, weiPerETH)
+		updatePrice(weiPrice)
+		return nil
+	}
+
+	watcher, err := watchers.NewPriceFeedWatcher(ctx, rpcURL, priceFeedAddr)
+	if err != nil {
+		return err
+	}
+
+	base, quote := watcher.Currencies()
+	if base != "ETH" && quote != "ETH" {
+		return fmt.Errorf("price feed does not have ETH as a currency (%v/%v)", base, quote)
+	}
+	if base != currency && quote != currency {
+		return fmt.Errorf("price feed does not have %v as a currency (%v/%v)", currency, base, quote)
+	}
+	updatePrice(priceDataToWei(pricePerUnit, watcher.Current(), base))
```

Comment:
What if price update fails here? Shouldn't we return an error (or at least some warning), so that someone knows that something can be misconfigured?

Reply:
There is no possibility of failure here! This config is just updated in memory as it used to be before. Do you think we are missing any safety checks here? One I had thought was protecting against 0/negative prices on the price feed, but that seemed like too much of a corner case.